Styra/Opa/OpaBatchTypes.cs

```diff
@@ -0,0 +1,74 @@
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Styra.Opa.OpenApi.Models.Components;
+
+namespace Styra.Opa;
+
+public class OpaBatchInputs : Dictionary<string, Dictionary<string, object>>
+{
+    public override string ToString()
+    {
+        return JsonConvert.SerializeObject(this);
+    }
+}
+
+public class OpaBatchResults : Dictionary<string, OpaResult>
+{
+    public override string ToString()
+    {
+        return JsonConvert.SerializeObject(this);
+    }
+}
+
+public class OpaBatchErrors : Dictionary<string, OpaError>
+{
+    public override string ToString()
+    {
+        return JsonConvert.SerializeObject(this);
+    }
+}
+
+// Used for converting inputs for the Batch Query API, and converting result types
+// into useful higher-level types.
+public static class DictionaryExtensions
+{
+    public static Dictionary<string, Input> ToOpaBatchInputRaw(this Dictionary<string, Dictionary<string, object>> inputs)
+    {
+        var opaBatchInputs = new Dictionary<string, Input>();
+        foreach (var kvp in inputs)
+        {
+            opaBatchInputs[kvp.Key] = Input.CreateMapOfAny(kvp.Value);
+        }
+        return opaBatchInputs;
+    }
+
+    public static OpaBatchErrors ToOpaBatchErrors(this Dictionary<string, Styra.Opa.OpenApi.Models.Errors.ServerError> errors)
+    {
+        var opaBatchErrors = new OpaBatchErrors();
+        foreach (var kvp in errors)
+        {
+            opaBatchErrors[kvp.Key] = new OpaError(kvp.Value);
+        }
+        return opaBatchErrors;
+    }
+
+    public static OpaBatchErrors ToOpaBatchErrors(this Dictionary<string, Styra.Opa.OpenApi.Models.Components.ServerError> errors)
+    {
+        var opaBatchErrors = new OpaBatchErrors();
+        foreach (var kvp in errors)
+        {
+            opaBatchErrors[kvp.Key] = new OpaError(kvp.Value);
+        }
+        return opaBatchErrors;
+    }
```

Comment:
Delete--only need one copy of a function. 😉

Reply:
Ah! This one is subtle. The two functions there *are* identical, *except* for the the exact `ServerError` type.
The two variants are:
 - `OpenApi.Models.Errors.ServerError`
 - `OpenApi.Models.Components.ServerError`

The existence of these variants is annoying, and a tripping hazard from Speakeasy. I'll ask them why the two variants exist-- I suspect it's due to the generator tripping up over something subtle in the [OpenAPI spec](https://github.com/StyraInc/enterprise-opa/tree/main/openapi) that the generator is basing stuff off of.

Your comment has nudged me to add explanatory comments for those functions in my latest changes, so that others don't get surprised by the extremely similar type names.

Reply:
Ah, touche. I did notice the difference in another area, but missed it here. 👍

Reply:
All good! It was something that could be missed at-a-glance, so it'd definitely trip up someone else in the future.